**Задача 4.\
В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. \
Какова вероятность, что он учится: \
a) на факультете A \
б) на факультете B \
в) на факультете C?**

**Решение:**\
Для решения этой задачи используем формулу Байеса: \
**P(S/E) = P(E/S) * P(S) / P(E)**, где \
S - студенты факультетов
E - событие, что сессия сдана


Пусть *n* - это кол-во студентов на факультете A, тогда на факультете B тоже учится *n* студентов, а на факультете C - *2n* студентов. Все студенты сдавали сессию, значит мы можем вычислить вероятность для каждого факультета:
1/n + 1/n + 1/(2 * n) = 1, отсюда <br>
**P(S<sub>A</sub>) = 0.25, P(S<sub>B</sub>) = 0.25, P(S<sub>C</sub>) = 0.5**

Найдем полную вероятность наступления события E, зная что P(ES<sub>A</sub>) = 0.8, P(ES<sub>A</sub>) = 0.7, P(ES<sub>A</sub>) = 0.9:\
**P(E) = P(ES<sub>A</sub>) * P(S<sub>A</sub>) + P(ES<sub>B</sub>) * P(S<sub>B</sub>) + P(ES<sub>C</sub>) * P(S<sub>C</sub>)**


In [3]:
PEsA = 0.8
PEsB = 0.7
PEsC = 0.9
PsA = PsB = 0.25
PsC = 0.5
PE = PEsA * PsA + PEsB * PsB + PEsC * PsC
print(f'Полная вероятность наступления события E равна: {PE: .4f}')


Полная вероятность наступления события E равна:  0.8250


a) Рассчитаем вероятность, что студент учится на факультете A:\
**P(S<sub>A</sub>E) = P(ES<sub>A</sub>) * P(S<sub>A</sub>) / P(E)**

In [12]:
PsAE = PEsA * PsA / PE
print(f'Вероятность факультета A равна: {PsAE: .4f}')

Вероятность факультета A равна:  0.2424


б) Рассчитаем вероятность, что студент учится на факультете B:\
**P(S<sub>B</sub>E) = P(ES<sub>B</sub>) * P(S<sub>B</sub>) / P(E)**

In [13]:
PsBE = PEsB * PsB / PE
print(f'Вероятность факультета A равна: {PsBE: .4f}')

Вероятность факультета A равна:  0.2121


в) Рассчитаем вероятность, что студент учится на факультете C:\
**P(S<sub>B</sub>E) = P(ES<sub>B</sub>) * P(S<sub>B</sub>) / P(E)**

In [14]:
PsCE = PEsC * PsC / PE
print(f'Вероятность факультета A равна: {PsCE: .4f}')

Вероятность факультета A равна:  0.5455


Ответ:\
Вероятность того, что студент учится на:\
- факультете A = 24%
- факультете B = 21%
- факультете C = 55%
